com.civicrm.hrjobroles/CRM/Hrjobroles/BAO/HrJobRoles.php

```diff
@@ -47,21 +47,21 @@ public static function getContactRoles($cuid)  {
   }
 
   /**
-   * Get options for a given job roles field along with their database IDs.
+   * Get option values for specific option group.
    *
    * @param String $fieldName
    *
-   * @return Array
+   * @return array
    */
   public static function buildDbOptions($fieldName) {
     $queryParam = array(1 => array($fieldName, 'String'));
-    $query = "SELECT cpv.id, cpv.label from civicrm_option_value cpv
+    $query = "SELECT cpv.value, cpv.label from civicrm_option_value cpv
               LEFT JOIN civicrm_option_group cpg on cpv.option_group_id = cpg.id
               WHERE cpg.name = %1";
     $options = array();
     $result = CRM_Core_DAO::executeQuery($query, $queryParam);
     while ($result->fetch()) {
-      $options[] =  array( 'id'=>$result->id, 'label'=>strtolower($result->label) );
+      $options[] =  array( 'value'=>$result->value, 'label'=>strtolower($result->label) );
     }
     return $options;
   }
```

Comment:
won't work ... to explain it to you let's assume that when you call :

`$optionValues = DAO_NAME::buidOptions('my_pseudo_constant');`

$optionValues will contain the following options :
`[ 'test1' => 'test 1' , 'test2' => 'test 2' ]` 

if you then via database insert you added a new option value to 'my_pseudo_constant' let us call it 'test3' and then you called 

`$optionValues = DAO_NAME::buidOptions('my_pseudo_constant');`  again 

... in this case $optionValues will still contain 

`[ 'test1' => 'test 1' , 'test2' => 'test 2' ]` 

and test 3 will not be added to the list ... the only for test3 to be added to the list is to add it via 
CRM_Core_BAO_OptionValue::create();  or CRM_Core_BAO_OptionValue::add() ... but CRM_Core_BAO_OptionValue::create() is a wrapper for CRM_Core_BAO_OptionValue::add()  with some extra bussnies logic ... so at the end both of them will call CRM_Core_PseudoConstant::flush();